cluster/images/provider-kafka/Makefile

```diff
@@ -1,28 +1,38 @@
 # ====================================================================================
 # Setup Project
 
-PLATFORMS := linux_amd64 linux_arm64
 include ../../../build/makelib/common.mk
 
 # ====================================================================================
 #  Options
-IMAGE = $(BUILD_REGISTRY)/provider-kafka-$(ARCH)
-OSBASEIMAGE = scratch
-include ../../../build/makelib/image.mk
+
+include ../../../build/makelib/imagelight.mk
 
 # ====================================================================================
 # Targets
 
 img.build:
 	@$(INFO) docker build $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--load" img.build.shared
+	@$(OK) docker build $(IMAGE)
+
+img.publish:
+	@$(INFO) docker publish $(IMAGE)
+	@$(MAKE) BUILD_ARGS="--push" img.build.shared
+	@$(OK) docker publish $(IMAGE)
+
+img.build.shared:
 	@cp Dockerfile $(IMAGE_TEMP_DIR) || $(FAIL)
-	@cp -R ../../../package $(IMAGE_TEMP_DIR) || $(FAIL)
-	@cd $(IMAGE_TEMP_DIR) && $(SED_CMD) 's|BASEIMAGE|$(OSBASEIMAGE)|g' Dockerfile || $(FAIL)
+	@cp -R ../../../package . $(IMAGE_TEMP_DIR) || $(FAIL)
 	@cd $(IMAGE_TEMP_DIR) && $(SED_CMD) 's|VERSION|$(VERSION)|g' package/crossplane.yaml || $(FAIL)
-	@cd $(IMAGE_TEMP_DIR) && find package -type f -name '*.yaml' -exec cat {} >> 'package.yaml' \; -exec printf '\n---\n' \; || $(FAIL)
-	@docker build $(BUILD_ARGS) \
-		--build-arg ARCH=$(ARCH) \
-		--build-arg TINI_VERSION=$(TINI_VERSION) \
+	@cd $(IMAGE_TEMP_DIR) && $(SED_CMD) 's|DOCKER_REGISTRY|$(DOCKER_REGISTRY)|g' package/crossplane.yaml || $(FAIL)
```

Comment:
Hmm we don't get `$(DOCKER_REGISTRY)` here anymore either.. the org is just embedded in `$(IMAGE)`, but we could update the `build` submodule to pass the org specifically... inserting `-controller` before `:` in the `$(IMAGE)` and replacing in `crossplane.yaml` would also work without changes

Reply:
Updated. Let me know if this setup works better.

Reply:
@stevendborrelli ah sorry I actually missed that this update won't work as is because `$(IMAGE)` includes the version so you'll have to do some more in-depth parsing to put `-controller` before the `:`. You could actually just do:
```
$ echo "crossplane/provider-kafka:v1.0.0" | sed 's/:/-controller:/'
crossplane/provider-kafka-controller:v1.0.0
```
And use that output to replace in the `package.yaml`

Reply:
I'm a bit confused. 

Are you sure `$(IMAGE)` includes the version? I get a value of something like: `build-aa9a2109/provider-kafka-arm64`.  I think in the  `img.build` target there is no version. https://github.com/upbound/build/blob/70b7766c9cc2e1f4fc36d5b8a2b3ffd4661a6df5/makelib/imagelight.mk#L104.

I just unpacked an image and it comes back with:
```
# make build VERSION=v1.2.3
---
apiVersion: meta.pkg.crossplane.io/v1alpha1
kind: Provider
metadata:
  name: provider-kafka
spec:
  controller:
    image: build-aa9a2109/provider-kafka-arm64-controller:v1.2.3
```

Reply:
@stevendborrelli ah good catch -- it is only set for publish, which uses this same shared logic.. the motivation for the structure we have is based on folks being able to use `IMAGE` and not being able to make a mistake, which doesn't play super nicely with this replace situation where we want to create a different image name based on the image name that is actually being built... there isn't a super clean way to do this, I looked at changing the `build` submodule and I feel like passing values that would make sense here would complicate the primary purpose. Furthermore, for the time-being we don't actually want to reference the `registry.upbound.io` controller image because Upbound Registry does not support anonymous pull of container images (only package images). All of that being said, I think the best (for the short term) solution here is just hard-coding:

```
spec:
  controller:
    image: crossplane/provider-kafka-controller:VERSION
```

Then doing replace of just the `VERSION`. Apologies for the long back and forth here!

Reply:
Updated.